wasm/module.go

```diff
@@ -8,50 +8,47 @@ import (
 	"github.com/tetratelabs/wazero/wasm/leb128"
 )
 
-var (
-	magic   = []byte{0x00, 0x61, 0x73, 0x6D}
-	version = []byte{0x01, 0x00, 0x00, 0x00}
+const (
+	magic   = "\x00asm"
+	version = "\x01\x00\x00\x00"
 )
 
-type Reader struct {
+type reader struct {
 	binary []byte
 	read   int
```

Comment:
ex this field is read directly. if we didn't implement `io.Reader` calls like `io.ReadFull` would fail. IOTW, we don't need to double-check an internal struct (IMHO).